File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        curr_max = 0

        for num in nums:
            curr_max = max(curr_max, 0) + num
            max_sum = max(max_sum, curr_max)
        return max_sum 
        

File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
     // map dan foydalanamiz. 

    //  agar mapni ichida bo'lsa false qaytarimiz bo'ldi. 
    let seen = new Map();

    for(let num of nums){
        if(seen.has(num)){
            return true
        }
      seen.set(num)
    }

    return false





    // for (let i = 0; i < nums.length; i++){
    //     for(let j = i + 1; j < nums.length; j++){
    //         if(nums[i] === nums[j]){
    //             return true
    //         }
    //     }
    // }
    // return false
    }
}
